src/compiler/compile/nodes/Element.ts

```diff
@@ -458,16 +465,39 @@ export default class Element extends Node {
 					should_have_attribute(this, required_attributes);
 				}
 			}
+		}
 
-			if (this.name === 'input') {
-				const type = attribute_map.get('type');
-				if (type && type.get_static_value() === 'image') {
-					const required_attributes = ['alt', 'aria-label', 'aria-labelledby'];
-					const has_attribute = required_attributes.some(name => attribute_map.has(name));
+		if (this.name === 'input') {
+			const type = attribute_map.get('type');
+			if (type && type.get_static_value() === 'image') {
+				const required_attributes = ['alt', 'aria-label', 'aria-labelledby'];
+				const has_attribute = required_attributes.some(name => attribute_map.has(name));
 
-					if (!has_attribute) {
-						should_have_attribute(this, required_attributes, 'input type="image"');
-					}
+				if (!has_attribute) {
+					should_have_attribute(this, required_attributes, 'input type="image"');
+				}
+			}
+		}
+
+		if (this.name === 'img') {
+			const alt_attribute = attribute_map.get('alt');
+			const aria_hidden_attribute = attribute_map.get('aria-hidden');
+
+			const aria_hidden_exist = aria_hidden_attribute && aria_hidden_attribute.get_static_value();
+			
+			if (alt_attribute && !aria_hidden_exist) {
+				const invalid_words = ['image', 'photo', 'picture'];
+				const alt_value = alt_attribute.get_static_value();
+
+				const contain_invalid_word = invalid_words.some(word => 
+					alt_value.toLowerCase().includes(word)
```

Comment:
I believe this would also complain about things like 'pilgrimage', 'photosynthesis', and 'picturesque'. In [the original plugin](https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/1a7b94f74ff1d565e2e2cc2db5adb06da26e98b3/src/rules/img-redundant-alt.js#L63), the logic is a bit more complicated, to avoid detecting parts of words.

Reply:
Hello @Conduitry, makes a lot of sense, I made this adjustment